### PR TITLE
Add dev to prod data replication script

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

/kind feature

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
Looks like you got some linter failures there 😄  they get us all.

```
build/scripts/site-config-version-update/main.go:73:3: ifElseChain: rewrite if-else to switch statement (gocritic)
		if strings.HasPrefix(line, "supported_k8s") {
		^
build/scripts/site-config-version-update/main.go:92:49: octalLiteral: use new octal literal style, 0o644 (gocritic)
	err = os.WriteFile(configFile, updatedContent, 0644)
	                                               ^
build/scripts/site-config-version-update/main.go:27:6: exported: exported type Config should have comment or be unexported (revive)
type Config struct {
     ^
build/scripts/site-config-version-update/main.go:16:1: package-comments: package comment is detached; there should be no blank lines between it and the package statement (revive)

^
build/scripts/site-config-version-update/main.go:28:12: fieldalignment: struct with 96 pointer bytes could be 88 (govet)
	DevConfig struct {
	          ^
build/scripts/site-config-version-update/main.go:37:13: fieldalignment: struct with 96 pointer bytes could be 88 (govet)
	ProdConfig struct {
	           ^
make: *** [Makefile:491: lint] Error 1
```

File: build/scripts/site-config-version-update/main.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+// Package main implements a program that replicates supported version of development to production.
+package main
+
+import (
+	"log"
+	"os"
+	"strings"
+
+	"github.com/pelletier/go-toml"
```
Comment (markmandel) [inline build/scripts/site-config-version-update/main.go:23]: Shouldn't there be a go.mod in here? Given this dependency?

File: docs/governance/templates/release_issue.md [hunk 2]
```diff
@@ -43,12 +43,7 @@ and copy it into a release issue. Fill in relevant values, found inside {}
   - [ ] config.toml updates:
     - [ ] Update `release_branch` to the new release branch for {version}.
     - [ ] Update `release_version` with the new release version {version}.
-    - [ ] Copy `dev_supported_k8s` to `supported_k8s`.
-    - [ ] Copy `dev_k8s_api_version` to `k8s_api_version`.
-    - [ ] Copy `dev_gke_example_cluster_version` to `gke_example_cluster_version`.
-    - [ ] Copy `dev_aks_example_cluster_version` to `aks_example_cluster_version`.
-    - [ ] Copy `dev_eks_example_cluster_version` to `eks_example_cluster_version`.
-    - [ ] Copy `dev_minikube_example_cluster_version` to `minikube_example_cluster_version`.
+    - [ ] Navigate to `agones` directory and run `go run build/scripts/site-config-version-update/main.go` to synchronizes data between dev and prod.
```
Comment (markmandel) [inline docs/governance/templates/release_issue.md:46]: Can we make this a `make` target in `website.mk` please - and then we can run the go code inside the build container as well.

I missed this in `go run build/scripts/version-update/main.go` above - but we should covert that over too (I'll make another bug for it).

**Review by markmandel — COMMENTED** [review]
Looks good!

File: build/scripts/site-config-version-update/main.go (hunk 1 above)
Comment (Kalaiselvi84) [reply build/scripts/site-config-version-update/main.go:23]: The go.mod and go.sum files are now in place.

File: docs/governance/templates/release_issue.md (hunk 2 above)
Comment (Kalaiselvi84) [reply docs/governance/templates/release_issue.md]: Created `site-config-update-version` target  in website.mk file.

**Comment by Kalaiselvi84** [comment]
Since I'm facing synchronization issues with my fork, I will delete the current branch and proceed [here](https://github.com/googleforgames/agones/pull/3224)

**Pull request closed without merging**